### DISCUSSION TITLE
some advise

### DISCUSSION THREAD

**Discussion by sddzcuigc** [kkebo/DNSecure#48] [Ideas]
1. add shortcuts method which help change dns for certain app autoly when open them 
2. Give examples of dns rules,I need to automaticly change dns for visiting different domain/ip,however I still not get the point.
3. Add whole opthion apple gives,  apple give an option for dns over ip,but our app only afford https and Tls.

**Comment by kkebo** [comment]
@sddzcuigc Thanks for your suggestions.

> 1. add shortcuts method which help change dns for certain app autoly when open them

Shortcuts support is a good idea, but do you have any use cases for this?

> 2. Give examples of dns rules,I need to automaticly change dns for visiting different domain/ip,however I still not get the point.

For example,

1. Change "Action" to "Do not apply settings"
2. Change "Interface Type" to "Wi-Fi"
3. Add the SSID on which you don't want to use DoT or DoH

In case of this example, while your device is connecting to the SSID that you specify, iOS uses the normal DNS settings instead of DNSecure.

![EFC2CC62-1AD2-4BA5-BE07-E2F41DA53862](https://user-images.githubusercontent.com/601636/131155943-55e768b2-fa54-45d6-80ff-42277b00447f.png)

> 3. Add whole opthion apple gives, apple give an option for dns over ip,but our app only afford https and Tls.

I believe that you can set it up in the Settings app without using DNSecure. If you don't mind me asking, in what cases would you want it?

**Comment by sddzcuigc** [comment]
I do have a adguardhome DNS server which help me do some personal dns work. 
I used to get my dns-setting change  by using the .mobileconfig file according to the https://developer.apple.com/documentation/devicemanagement/dnssettings/
I can use dns-setting profile1 with some payloads and use dns-setting porfile2 with other payloads which can do the work DNSecure app does!
Howerver,
some time I have to block ads  on some app using the dns-setting profile1,
other time I have to visit the local area network use some domain I make on some other app using the dns-setting profile2,
and other time I have to stop using the personal dns-setting profiles and turn it to “auto”
I deadly need it automaticly changed among my dns-setting profiles when I open some certain app, I found the  shortcuts app give methods to do something when open certain app.But it don't give methods to change dns-setting profiles. I think it is a good idea to make a app to do this , but I don't have some programming skills which you great guys have. 
**Thanks a lot!**

**Comment by sddzcuigc** [comment]
It will be like this:
When using the shortcuts app, I set it when open app 1 >it open DNSecure and change its profile to profile1.
When suing the shortcuts app, I set it when open app 2 >it open DNSecure and change its profile to profile2.
This may need DNSecure give a api to shortcuts.
I do find some app give its inner link to shortcuts.,maybe using scheme link. I say maybe because I don't understand programing and the ios develop at all.
@kkk669

**Comment by sddzcuigc** [comment]
And you said "I believe that you can set it up in the Settings app without using DNSecure. If you don't mind me asking, in what cases would you want it?" I used to change the dns while using the setting app ，but it won't work with the cell phone.

**Comment by kkebo** [comment]
> I used to change the dns while using the setting app ，but it won't work with the cell phone.

I understand. You were talking about the case of cellular networks.

But unfortunately, Apple doesn't give us an option for that. According to https://developer.apple.com/documentation/networkextension/nednssettingsmanager/3552342-dnssettings, DoH or DoT seems to be the only choice.

Anyway, this issue looks like feature requests, so I'll convert it into a discussion.
